src/astro/databases/snowflake.py

```diff
@@ -57,6 +176,121 @@ def get_table_qualified_name(table: Table) -> str:  # skipcq: PYL-R0201
         qualified_name = ".".join(name for name in qualified_name_lists if name)
         return qualified_name
 
+    # ---------------------------------------------------------
+    # Snowflake stage methods
+    # ---------------------------------------------------------
+
+    @staticmethod
+    def _create_stage_auth_sub_statement(
+        file: File, storage_integration: Optional[str] = None
+    ) -> str:
+        """
+        Create authentication-related line for the Snowflake CREATE STAGE.
+        Raise an exception if it is not defined.
+
+        :param file: File to be copied from/to using stage
+        :param storage_integration: Previously created Snowflake storage integration
+        :return: String containing line to be used for authentication on the remote storage
+        """
+
+        if storage_integration is not None:
+            auth = f"storage_integration = {storage_integration};"
+        else:
```

Comment:
@tatiana Would it make sense to simplify this function? to have a different function to perform checks for S3 and GCS? WDYT?